test/utils_test.cpp

```diff
@@ -27,7 +28,18 @@ std::string getTmpFileName() {
   if (user != nullptr) {
     userstr = std::string(user);
   }
-  return std::string("/tmp/test_") + userstr + std::string(".graph");
+  // tmp dir
+  char* dir1 = std::getenv("TMPDIR");
+  std::string tmpdir = "";
+  if (dir1 != nullptr) {
+    tmpdir = std::string(dir1);
+  }
+  char* dir2 = std::getenv("TMP");
+  if (dir2 != nullptr) {
+    tmpdir = std::string(dir2);
+  }
+
+  return tmpdir + "_test" + userstr + ".graph";
```

Comment:
I think we need to add append an equivalent of `os.sep` to `tmpdir` for concatenating. 

Also, from the link, I see `Other forms sometimes accepted are TEMP, TEMPDIR and TMP`, but we are not testing `TEMP`.

Reply:
fair enough - `TMPDIR` and `TMP` covers *nix-based and Windows systems, but I'll add `TEMP` too